### DISCUSSION TITLE
Information on OrchardCore.Secrets module

### DISCUSSION THREAD

**Discussion by DotCat1985** [OrchardCMS/OrchardCore#11173] [General]
When will **OrchardCore.Secrets** module be released?

[OrchardDojo](https://orcharddojo.net/blog/secret-management-updates-challengeorforbid-in-api-controllers-this-week-in-orchard-18-09-2020)
[GitHub Dean Marcussen > secrets branch](https://github.com/OrchardCMS/OrchardCore/tree/deanmarcussen/secrets/src/OrchardCore.Modules/OrchardCore.Secrets)

I think it's useful to encrypt stmp password.

**Comment by hishamco** [comment]
@Tokyo1985 the SMTP passwords are encrypterd today via DataProtection APIs. @deanmarcussen introduced a notion of OC.Secrets long time back, I think it's WIP. So let us wait for his comment and the advantage over the current protection

**Reply by DotCat1985** [reply]
OK, how can I encrypt my plain SMTP password inside recipe or anything?

**Reply by hishamco** [reply]
I didn't try it in the recipes, @deanmarcussen @ns8482e @Skrypt did anyone of you encrypt the SMTP password in the recipes

> anything?

You could use Data Protection APIs

**Reply by DotCat1985** [reply]
> You could use Data Protection APIs
In which class?

**Reply by hishamco** [reply]
You can use `IDataProtectionProvider`, here's a sample usage

https://github.com/OrchardCMS/OrchardCore/blob/556359c75bf6ab3343b3343d6bcc4541c1b7bce1/src/OrchardCore.Modules/OrchardCore.Email/Drivers/SmtpSettingsDisplayDriver.cs#L94-L96

**Reply by DotCat1985** [reply]
> You can use `IDataProtectionProvider`, here's a sample usage
> 
> https://github.com/OrchardCMS/OrchardCore/blob/556359c75bf6ab3343b3343d6bcc4541c1b7bce1/src/OrchardCore.Modules/OrchardCore.Email/Drivers/SmtpSettingsDisplayDriver.cs#L94-L96

OK, thank you

**Reply by Skrypt** [reply]
I think the idea was to use the Secret module that @deanmarcussen was working on. @deanmarcussen seems busy lately. DataProtection is another way to do this.

**Reply by DotCat1985** [reply]
> I think the idea was to use the Secret module that @deanmarcussen was working on. @deanmarcussen seems busy lately. DataProtection is another way to do this.
OK, I solved the issue using your solution. Thank you!
